core/dbt/task/compile.py

```diff
@@ -43,19 +43,32 @@ def raise_on_first_error(self):
         return True
 
     def get_node_selector(self) -> ResourceTypeSelector:
+        if getattr(self.args, "inline", None):
+            resource_types = [NodeType.SqlOperation]
+        elif getattr(self.args, "select", None):
+            resource_types = [NodeType.Model]
```

Comment:
Discussed offline yesterday with @aranke @ChenyuLInx:

This is to handle the fact that `--select my_model` will also select the tests on `my_model`, and we really only want to be compiling one node. The right way to handle this behavior is to turn off "indirect selection," and the cleanest approach there is probably to introduce a new option to turn off indirect selection entirely (`--indirect-selection empty`).

Depending on how much work is involved there, we can scope & estimate that as a separate issue.

Reply:
Did y'all consider any alternative names to `empty`? For example, `none` feels more intuitive to me, but maybe y'all considered it already.

Adding `empty` to the [current options](https://docs.getdbt.com/reference/node-selection/test-selection-examples#indirect-selection) would give:
1. `eager` (default) - include ANY test that references the selected nodes
1. `cautious` - restrict to tests that ONLY refer to selected nodes
1. `buildable` - restrict to tests that ONLY refer to selected nodes (or their ancestors)
1. `empty` - don't include any tests

If it is named something like `none`, then the list would be this instead:
1. `eager`
1. `cautious`
1. `buildable`
1. `none`